gr-iio/lib/ad9081_common_impl.cc

```diff
@@ -0,0 +1,106 @@
+/* -*- c++ -*- */
+/*
+ * Copyright 2021 Free Software Foundation, Inc.
+ *
+ * This file is part of GNU Radio
+ *
+ * SPDX-License-Identifier: GPL-3.0-or-later
+ *
+ */
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include "ad9081_common_impl.h"
+
+#include <gnuradio/iio/ad9081_common.h>
+
+#include <cmath>
+#include <cstdint>
+
+namespace gr {
+namespace iio {
+
+long long phase_fold(float x)
+{
+    return (long long)(1000 *
+                       (x - static_cast<int64_t>(floor((x + 180.0) / 360.0)) * 360.0));
+}
+
+int64_t nyquist_fold(int64_t f_s, int64_t f_c /*, int64_t& f_lo, bool& odd*/)
+{
+    int64_t f_n = f_s >> 1;
```

Comment:
*cough*

```c++
#include "fmt/core.h"
#include <cstdint>
int main(){
  int64_t neg_one = -1;
  int64_t result = neg_one >> 1;
  fmt::print("neg_one={:d}, neg_one >> 1={:d}\n", neg_one, result);
  int64_t neg_two = -2;
  int64_t result_two = neg_two >> 1;
  fmt::print("neg_two={:d}, neg_two >> 1={:d}\n", neg_two, result_two);
  int64_t pos_one = 1;
  int64_t result_pos = pos_one >> 1;
  fmt::print("pos_one={:d}, pos_one >> 1={:d}\n", pos_one, result_pos);
}
```

yields
```
    /tmp  clang++ -o test -Wall -Wpedantic -lfmt int64.cc                                                                                                                                                                                                    ✔  base 🐍  00:00:19 
    /tmp  ./test                                                                                                                                                                                                                                             ✔  base 🐍  00:00:57 
neg_one=-1, neg_one >> 1=-1
neg_two=-2, neg_two >> 1=-1
pos_one=1, pos_one >> 1=0
```

Is `f_s` really meant to be an `int64_t` argument? In that case, use `/2` instead of `>>1`.

Reply:
To be fair, it works for all significant inputs, but i agree i should let the compiler do the shifting :P